src/ethereum_fmt_plug.c

```diff
@@ -0,0 +1,241 @@
+/*
+ * JtR format to crack password protected Ethereum Wallets.
+ *
+ * This software is Copyright (c) 2017, Dhiru Kholia <kholia at kth.se> and it
+ * is hereby released to the general public under the following terms:
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted.
+ */
+
+#if FMT_EXTERNS_H
+extern struct fmt_main fmt_ethereum;
+#elif FMT_REGISTERS_H
+john_register_one(&fmt_ethereum);
+#else
+
+#include <string.h>
+#ifdef _OPENMP
+#include <omp.h>
+#ifndef OMP_SCALE
+#define OMP_SCALE               1
+#endif
+#endif
+
+#include "arch.h"
+#include "misc.h"
+#include "common.h"
+#include "formats.h"
+#include "params.h"
+#include "options.h"
+#include "pbkdf2_hmac_sha256.h"
+#include "ethereum_common.h"
+#include "escrypt/crypto_scrypt.h"
+#include "KeccakHash.h"
+#include "jumbo.h"
+#include "memdbg.h"
+
+#define FORMAT_NAME             "Ethereum Wallet"
+#define FORMAT_LABEL            "ethereum"
+#ifdef SIMD_COEF_64
+#define ALGORITHM_NAME          "PBKDF2-SHA256 " SHA1_ALGORITHM_NAME
+#else
+#define ALGORITHM_NAME          "PBKDF2-SHA256 32/" ARCH_BITS_STR
```

Comment:
Thanks!

```
diff --git a/src/ethereum_fmt_plug.c b/src/ethereum_fmt_plug.c
index 43e1946..089527e 100644
--- a/src/ethereum_fmt_plug.c
+++ b/src/ethereum_fmt_plug.c
@@ -38,9 +38,9 @@ john_register_one(&fmt_ethereum);
 #define FORMAT_NAME             "Ethereum Wallet"
 #define FORMAT_LABEL            "ethereum"
 #ifdef SIMD_COEF_64
-#define ALGORITHM_NAME          "PBKDF2-SHA256 " SHA1_ALGORITHM_NAME
+#define ALGORITHM_NAME          "PBKDF2-SHA256/scrypt Keccak " SHA256_ALGORITHM_NAME
 #else
-#define ALGORITHM_NAME          "PBKDF2-SHA256 32/" ARCH_BITS_STR
+#define ALGORITHM_NAME          "PBKDF2-SHA256/scrypt Keccak 32/" ARCH_BITS_STR
 #endif
 #define BENCHMARK_COMMENT       ""
 #define BENCHMARK_LENGTH        -1
```

This results in the following output,
```
Benchmarking: ethereum, Ethereum Wallet [PBKDF2-SHA256/scrypt Keccak 256/256 AVX2 8x]...
Speed for cost 1 (iteration count) of 262144 and 1024
Raw:	62.1 c/s real, 47.4 c/s virtual
```

However `AVX2` is not correct for `scrypt` I think.